### ISSUE TITLE
`brew install elixir` installs erlang 19 even if erlang 18 requirement is satisfied

### ISSUE THREAD

**Issue by elliotthilaire** [Homebrew/homebrew-core#10603]
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [ ] Confirmed this is a problem with specific formulae and not Homebrew/brew? If it's a general Homebrew/brew problem please file this issue at https://github.com/Homebrew/brew/issues/new

I am unsure how to confirm if this is a Homebrew/brew issue or related to this particular formula.
It may be a bug with Requirement, satisfy and default_formula.

### Bug reports:

`brew install elixir` tries to install erlang formula with v19 even when Erlang18Requirement#satisfy is true.

1. Install erlang v18 with `brew install  erlang@18`
2. Add it to PATH (because it is keg only)
3. Confirm `erl` is version 18
    ```
     Erlang/OTP 18 [erts-7.3.1] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]
    ```
4.  Run `brew install elixir` and observe that `erlang` formula is still installed
    ```
    ==> Installing dependencies for elixir: erlang
    ==> Installing elixir dependency: erlang
    ==> Downloading https://homebrew.bintray.com/bottles/erlang-19.2.3.sierra.bottle
    Already downloaded: /Users/elliott/Library/Caches/Homebrew/erlang-19.2.3.sierra.bottle.tar.gz
    ==> Pouring erlang-19.2.3.sierra.bottle.tar.gz
    ==> Caveats
    ...
    ```

I tried replacing Erlang18Requirement#satisfy with 
```
satisfy do 
   true
end
```
and erlang formula with v19 is still installed with elixir despite already having 18.

## brew config
```
HOMEBREW_VERSION: 1.1.10-454-ga1da909de
ORIGIN: https://github.com/Homebrew/brew
HEAD: a1da909deaf13f94f83f5885693457898d6d49e1
Last commit: 82 minutes ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 824a8cb7c75d555c7ece4988b12bb1e75fb2fb37
Core tap last commit: 16 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: quad-core 64-bit ivybridge
Homebrew Ruby: 2.0.0-p648
Clang: 8.0 build 800
Git: 2.12.0 => /usr/local/bin/git
Perl: /usr/local/bin/perl => /usr/local/Cellar/perl/5.24.1/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /Users/elliott/.rbenv/shims/ruby => /Users/elliott/.rbenv/versions/2.4.0/bin/ruby
Java: 1.8.0_92
macOS: 10.12.3-x86_64
Xcode: 8.2.1
CLT: 8.2.0.0.1.1480973914
X11: N/A
```

## brew doctor
```
Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libwkhtmltox.0.12.3.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/wkhtmltox/image.h
  /usr/local/include/wkhtmltox/pdf.h
```

**Comment by ilovezfs** [comment]
This seems to work:
```
diff --git a/Formula/elixir.rb b/Formula/elixir.rb
index c644942..d3c8056 100644
--- a/Formula/elixir.rb
+++ b/Formula/elixir.rb
@@ -3,10 +3,10 @@ class Erlang18Requirement < Requirement
   default_formula "erlang"
 
   satisfy do
-    erl = which("erl")
+    erl = which("erl") || which(Formula["erlang@18"].opt_bin/"erl")
     next unless erl
     `#{erl} -noshell -eval 'io:fwrite("~s", [erlang:system_info(otp_release) >= "18"])' -s erlang halt | grep -q '^true'`
-    $?.exitstatus.zero?
+    $?.exitstatus.zero? ? erl : false
   end
 
   def message; <<-EOS.undent
```

**Comment by elliotthilaire** [comment]
You are correct. There are two problems which you have solved.

1. It appears that we have to return a path, i.e. the result of `which("erl")` as you have done with `$?.exitstatus.zero? ? erl : false`
Simply returning `true` is not satisfactory, and will cause the default_formula to be installed regardless.

2. `brew install erlang@18` is not enough to satisfy on its own. It needs to be in PATH too so that which("erl") returns something.
brew tells us how to do this: `echo 'export PATH="/usr/local/opt/erlang@18/bin:$PATH"' >> ~/.bash_profile`
Forgetting to reload .bash_profile before running `brew install elixir` will install erlang 19

I don't think we should add `|| which(Formula["erlang@18"].opt_bin/"erl")` because elixir will fail to run if erl is not in PATH.

In summary:
We should add `$?.exitstatus.zero? ? erl : false`

Then:

This will still not work:
```
brew install erlang@18
brew install elixir
```

But this will:
```
brew install erlang@18
`echo 'export PATH="/usr/local/opt/erlang@18/bin:$PATH"' >> ~/.bash_profile`
source ~/.bash_profile
brew install elixir
```

**Comment by ilovezfs** [comment]
Sounds good to me.

**Comment by ilovezfs** [comment]
Thanks for reporting this!
